plat/rockchip/rk3399/drivers/dp/cdn_dp.c

```diff
@@ -41,3 +44,46 @@ __asm__(
 	".popsection\n"
 );
 
+static uint64_t *hdcp_key_pdata;
+static struct cdn_dp_hdcp_key_1x key;
+
+int hdcp_handler(struct cdn_dp_hdcp_key_1x *key);
+
+uint64_t dp_hdcp_ctrl(uint64_t type)
+{
+	switch (type) {
+	case HDCP_KEY_DATA_START_TRANSFER:
+		memset(&key, 0x00, sizeof(key));
+		hdcp_key_pdata = (uint64_t *)&key;
+		return 0;
+	case HDCP_KEY_DATA_START_DECRYPT:
+		if (hdcp_key_pdata == (uint64_t *)(&key + 1))
```

Comment:
Yup, that should be better for me. 

For example:

+/* Checks the cdn_dp_hdcp_key_1x must be aligned on 6 x 64-bit word boundary */
 +CASSERT(sizeof(struct cdn_dp_hdcp_key_1x) % HDCP_KEY_1X_STORE_DATA_ALIGN_SIZE, \
 +	assert_hdcp_key_1x_store_data_align_size_mismatch);